rust/scx_loader/src/main.rs

```diff
@@ -139,11 +149,73 @@ impl ScxLoader {
     }
 }
 
+// Monitors CPU utilization and enables scx_lavd when utilization of any CPUs is > 90%
+async fn monitor_cpu_util() -> Result<()> {
+    let mut system = System::new_all();
+    let mut running_sched: Option<Child> = None;
+    let mut cpu_above_threshold_since: Option<Instant> = None;
+    let mut cpu_below_threshold_since: Option<Instant> = None;
+    
+    let high_utilization_threshold = 90.0;
+    let low_utilization_threshold_duration = Duration::from_secs(30);
+    let high_utilization_trigger_duration = Duration::from_secs(5);
+    
+    loop {
+        system.refresh_cpu_all();
+        
+        let any_cpu_above_threshold = system.cpus().iter().any(|cpu| cpu.cpu_usage() > high_utilization_threshold);
+        
+        if any_cpu_above_threshold {
+            if cpu_above_threshold_since.is_none() {
+                cpu_above_threshold_since = Some(Instant::now());
+            }
+            
+            if cpu_above_threshold_since.unwrap().elapsed() > high_utilization_trigger_duration {
+                if running_sched.is_none() {
+                    println!("CPU Utilization exceeded 90% for 5 seconds, starting scx_lavd");
+                    running_sched = Some(std::process::Command::new("scx_lavd").spawn().expect("Failed to start scx_lavd"));
```

Comment:
1. Process spawning from tokio should used here instead. To not mix API and have it consistent
2. I think Enum to string conv should be used here instead of hardcoding